apps/settings/js/bootstrap.js

```diff
@@ -1,41 +1,78 @@
-// A factory to load panels in an iframe.
-class IframePanelFactory {
-  log(msg) {
-    console.log(`IframePanel: ${msg}`);
-  }
+function elem(id) {
+  return document.getElementById(id);
+}
 
-  open(name) {
-    if (!this.iframe) {
-      this.iframe = document.createElement("iframe");
-      this.iframe.setAttribute("id", "panel");
-      document.body.appendChild(this.iframe);
-    }
-    this.iframe.contentWindow.location.replace(`./panels/${name}/index.html`);
-    this.log(`loading panel from ./panels/${name}/index.html`);
-    this.iframe.classList.add("open");
-  }
+const bootstrapEvent = new CustomEvent('bootstrap', {});
 
-  hide() {
-    if (this.iframe) {
-      this.iframe.classList.remove("open");
-      this.iframe.src = "about:blank";
-    }
+async function processLocation() {
+  let sections = this.document.querySelectorAll("section");
+  sections.forEach(i => {
+    i.classList.add("hidden");
+  })
+  if (window.location.hash === "") {
+    elem("main").classList.remove("hidden");
+  } else {
+    let name = window.location.hash.substring(1);
+    elem("spinner").classList.remove("hidden");
+    let e = elem(name);
```

Comment:
nit: please avoid single letter variables.

What happens if the hash does not match an existing panel? We should fallback on the main screen.

Reply:
Replaced `e` with `panel` and implemented a fall-back to main menu.